homeassistant/components/tradfri/light.py

```diff
@@ -45,56 +52,81 @@ async def async_setup_entry(
 ) -> None:
     """Load Tradfri lights based on a config entry."""
     gateway_id = config_entry.data[CONF_GATEWAY_ID]
-    tradfri_data = hass.data[DOMAIN][config_entry.entry_id]
-    api = tradfri_data[KEY_API]
-    devices = tradfri_data[DEVICES]
-
-    entities: list[TradfriBaseClass] = [
-        TradfriLight(dev, api, gateway_id) for dev in devices if dev.has_light_control
+    coordinator_data = hass.data[DOMAIN][config_entry.entry_id][COORDINATOR]
+    api = coordinator_data[KEY_API]
+
+    entities: list = [
+        TradfriLight(
+            device_coordinator,
+            api,
+            gateway_id,
+        )
+        for device_coordinator in coordinator_data[COORDINATOR_LIST]
+        if device_coordinator.device.has_light_control
     ]
-    if config_entry.data[CONF_IMPORT_GROUPS] and (groups := tradfri_data[GROUPS]):
-        entities.extend([TradfriGroup(group, api, gateway_id) for group in groups])
+
+    if config_entry.data[CONF_IMPORT_GROUPS] and (
+        group_coordinators := coordinator_data[GROUPS_LIST]
+    ):
+        entities.extend(
+            [
+                TradfriGroup(group_coordinator, api, gateway_id)
+                for group_coordinator in group_coordinators
+            ]
+        )
+
     async_add_entities(entities)
 
 
-class TradfriGroup(TradfriBaseClass, LightEntity):
+class TradfriGroup(CoordinatorEntity, LightEntity):
     """The platform class for light groups required by hass."""
 
     _attr_supported_features = SUPPORTED_GROUP_FEATURES
 
     def __init__(
         self,
-        device: Command,
+        group_coordinator: TradfriGroupDataUpdateCoordinator,
         api: Callable[[Command | list[Command]], Any],
         gateway_id: str,
     ) -> None:
         """Initialize a Group."""
-        super().__init__(device, api, gateway_id)
+        super().__init__(coordinator=group_coordinator)
 
-        self._attr_unique_id = f"group-{gateway_id}-{device.id}"
-        self._attr_should_poll = True
-        self._refresh(device, write_ha=False)
+        self._group: Group = self.coordinator.data
 
-    async def async_update(self) -> None:
-        """Fetch new state data for the group.
+        self._api = api
+        self._attr_unique_id = f"group-{gateway_id}-{self._group.id}"
+
+    @callback
+    def _refresh(self) -> None:
+        """Refresh the group."""
+        self._group = self.coordinator.data
```

Comment:
The group instance should be the same all the time. I don't think we need to set the group after every refresh.